scipy/special/cephes/hyp2f1.c

```diff
@@ -83,6 +83,43 @@ static double hyt2f1(double a, double b, double c, double x, double *loss);
 static double hys2f1(double a, double b, double c, double x, double *loss);
 static double hyp2f1ra(double a, double b, double c, double x,
 		       double *loss);
+static double pochhammer(double x, double n);
+static double factorial(double k);
+static double hyp2f1_indeterminate(double a, double b, double x);
+
+/*
+    See http://mathworld.wolfram.com/PochhammerSymbol.html
+*/
+static double pochhammer(double x, double n)
+{
+    return gamma(x + n) / gamma(x);	
+}
+
+/*
+    See http://mathworld.wolfram.com/GammaFunction.html
+*/
+static double factorial(double k)
+{
+    return gamma(k + 1);
+}
+
+/*
+    See https://math.stackexchange.com/questions/2457831/hypergeometric-expansion-approximation
+*/
+static double hyp2f1_indeterminate(double a, double b, double x)
+{
+    /*
+    	"converting floating point to integers is shockingly expensive"
+    */
+    double k;
+    double sum = 0;
+
+    for (k = 0; k <= -b; k++) {
+	sum += pochhammer(a, k) * pow(x, k) / factorial(k);
```

Comment:
Ideally, you would also calculate the error estimate for the sum (relevant for negative `x` when it is alternating) and return nan on total precison loss.
Unfortunately, the rest of hyp2f1 is not written as carefully.